### PR TITLE
feat: add tabbed navigation for development site

### LINKED ISSUE #288: Enable tabbed navigation per component
see  comp - end goal is we should be able to navigate from:
A. examples page with multiple examples of a component
B. Detail pages with documentation, accessibility, etc

This task should provide the mechanism to create multiple views per component and add tabbed interface to app-bar.

### PR DESCRIPTION
**Pull request by nicholasrice**
closes #288

### CODE REVIEW

File: packages/fast-development-site-react/src/components/site/index.tsx [hunk 1]
```diff
@@ -93,87 +95,125 @@ class Site extends React.Component<ISiteProps & IManagedClasses<ISiteManagedClas
...
                     {this.getPaneCollapseToggle()}
-                    {this.getSlotItems(this, ShellSlot.pane)}
+                    {this.getChildrenBySlot(this, ShellSlot.pane)}
                     <ul className={this.props.managedClasses.site_paneToc}>
-                        {this.getRootToc(this.props.children, SiteSlot.category, path, "/")}
+                        {this.getRootToc(this.props.children, SiteSlot.category, route.route, "/")}
                     </ul>
                 </ShellPane>
-                <ShellCanvas>
+                <ShellCanvas >
```
Comment (chrisdholt) [inline packages/fast-development-site-react/src/components/site/index.tsx]: nit: seems like an errant space

File: packages/fast-development-site-react/src/components/site/index.tsx [hunk 2]
```diff
@@ -93,87 +95,125 @@ class Site extends React.Component<ISiteProps & IManagedClasses<ISiteManagedClas
...
             </ShellInfoBar>
         );
     }
 
     /**
      * Gets all of the potential routes as strings to be used to build the shell
      */
-    private getRoutes(items: JSX.Element, baseRoute: string, slot: string, routes?: IComponentRoutes[]): IComponentRoutes[] {
-        let currentRoutes: IComponentRoutes[] = routes;
+    private getRoutes(items: JSX.Element, baseRoute: string, slot: SiteSlot, routes?: IComponentRoute[]): IComponentRoute[] {
```
Comment (chrisdholt) [inline packages/fast-development-site-react/src/components/site/index.tsx]: Do we need to check that routes exist here?

File: packages/fast-development-site-react/src/components/site/action-bar.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+                <div className={this.props.managedClasses.actionBar_componentViewToggles}>
+                    <ComponentViewToggle
+                        to={this.props.match.path}
+                        onClick={this.onComponentViewChangeCallback(ComponentViewTypes.detail)}
+                        label="View detail"
+                        current={this.isAriaCurrent(ComponentViewTypes.detail)}
+                        glyph={glyphPage}
+                    />
+                    <ComponentViewToggle
+                        to={this.props.match.path + `${ComponentViewTypes[ComponentViewTypes.examples]}/`}
```
Comment (janechu) [inline packages/fast-development-site-react/src/components/site/action-bar.tsx]: could stick the this.props.match.path in with the backticks to do  \`${this.props.match.path}${ComponentViewTypes[ComponentViewTypes.examples]}/\`

File: packages/fast-development-site-react/src/components/site/index.tsx (hunk 2 above)
Comment (nicholasrice) [reply packages/fast-development-site-react/src/components/site/index.tsx]: na, it is handled if it is undefined

**Pull request merged**